### ISSUE TITLE
bug: Unable to simplify formula due to an error | t2.evaluateSync is not a function

### ISSUE THREAD

**Issue by kgar** [kgar/foundry-vtt-tidy-5e-sheets#655]
When preparing action items with a spell like Magic Missile, I am getting an error in Foundry V12 and Tidy V4.4.1:

```
logging.ts:42 tidy5e-sheet | Unable to simplify formula due to an error. TypeError: t2.evaluateSync is not a function
    at formula.ts:57:19
    at Array.map (<anonymous>)
    at simplifyFormula (formula.ts:53:40)
    at actions.ts:235:23
    at Array.map (<anonymous>)
    at mapActionItem (actions.ts:227:40)
    at actions.ts:74:30
    at Array.map (<anonymous>)
    at getActorActionSections (actions.ts:74:8)
    at Tidy5eCharacterSheet.getData (Tidy5eCharacterSheet.ts:756:16)
    at async Tidy5eCharacterSheet._renderSheet (Tidy5eCharacterSheet.ts:1535:18)
    at async Tidy5eCharacterSheet.ts:1529:13
    at async mutex.ts:26:9
```

![image](https://github.com/kgar/foundry-vtt-tidy-5e-sheets/assets/4632129/76e02bf6-ef19-4133-8df7-15fe0f59a2d2)

Most likely, the version check logic is backwards. Verify by running the same thing in V11.

**Comment by kgar** [comment]
Haven't seen it since, and we've had some changes to this set of functions. Closing this.

**Comment by Belodri** [comment]
Getting this same error in Foundry V12 Build 330, dnd5e 3.3.1 and Tidy 5.4.4 when creating a weapon with this damage formula `1d(@prof) + (2 * @prof)`.

```
logging.ts:79  tidy5e-sheet | Unable to simplify formula due to an error. TypeError: t2.evaluateSync is not a function
    at formula.ts:57:19
    at Array.map (<anonymous>)
    at simplifyFormula (formula.ts:53:40)
    at actions.ts:215:23
    at Array.map (<anonymous>)
    at mapActionItem (actions.ts:207:40)
    at getActorActionSections (actions.ts:54:32)
    at async Tidy5eCharacterSheet.getData (Tidy5eCharacterSheet.ts:724:16)
    at async Tidy5eCharacterSheet._renderSheet (Tidy5eCharacterSheet.ts:1373:18)
    at async Tidy5eCharacterSheet.ts:1365:13
    at async mutex.ts:26:9
```

There's also a different error that I assume to be related which happens when the damage of a weapon is set to `1d(@prof)` or `1d@prof`. Both errors only happen if the weapon is on the action list.

```
tidy5e-sheet | Unable to simplify formula due to an error. TypeError: Cannot read properties of undefined (reading 'name')
    at DiceTerm.fromParseNode (foundry-esm.js:61739:61)
    at foundry-esm.js:63891:20
    at Array.map (<anonymous>)
    at Roll.instantiateAST (foundry-esm.js:63889:50)
    at Roll.parse (foundry-esm.js:63878:19)
    at new Roll (foundry-esm.js:63222:37)
    at Roll.create (foundry-esm.js:63698:14)
    at simplifyFormula (formula.ts:51:23)
    at actions.ts:215:23
    at Array.map (<anonymous>)
    at mapActionItem (actions.ts:207:40)
    at getActorActionSections (actions.ts:54:32)
    at async Tidy5eCharacterSheet.getData (Tidy5eCharacterSheet.ts:724:16)
    at async Tidy5eCharacterSheet._renderSheet (Tidy5eCharacterSheet.ts:1373:18)
    at async Tidy5eCharacterSheet.ts:1365:13
    at async mutex.ts:26:9
```

Apart from spamming the log, all of these work just fine and there are no issues with the item being displayed on the sheet either.

**Comment by kgar** [comment]
@Belodri thank you for the report. I was able to reliably recreate the issue from your steps. I should be able to get a fix out with the next Tidy release.

**Comment by kgar** [comment]
This has been released in Tidy 5.4.5. Hoping we've seen the last of it 🤞
